src/components/Typography/H1.jsx

```diff
@@ -1,18 +1,37 @@
 import React from "react";
 import PropTypes from "prop-types";
 import styles from "./styles.css";
-import { classNames } from "../../utils";
+import { classNames, idPropTypeCheckForTruncatedComponents } from "../../utils";
+import { useTruncate } from "../../hooks";
 
-const H1 = ({ children, className, style }) => (
-  <h1 className={classNames(styles.header1, className)} style={style}>
-    {children}
-  </h1>
-);
+const H1 = ({
+  id,
+  children,
+  className,
+  style,
+  truncate = false,
+  linesBeforeTruncate = 1
+}) => {
+  useTruncate(id, truncate, linesBeforeTruncate);
+  return (
+    <h1 id={id} className={classNames(styles.header1, className)} style={style}>
+      {children}
+    </h1>
+  );
+};
 
 H1.propTypes = {
   children: PropTypes.node,
   className: PropTypes.string,
-  style: PropTypes.object
+  style: PropTypes.object,
+  truncate: PropTypes.bool,
+  linesBeforeTruncate: PropTypes.number,
+  id: idPropTypeCheckForTruncatedComponents
+};
+
+H1.defaultProps = {
```

Comment:
You don't need defaultProps if you are using object destructuring in the component props. Choose one or another.